test/test_ripper_tags.rb

```diff
@@ -76,6 +76,39 @@ def imethod5
     ], tags.map{ |t| t[:full_name] }
   end
 
+  def test_nested_constant_definitions
+    tags = extract(<<-EOC)
+      A = [
+        B = 1,
+        C = 2
+      ]
+
+      D = {
+        E = 3 => F = 4,
+        x: G = 5
+      }
+
+      H = []
+
+      I = [
+        J = [
+          K = 1
+        ]
+      ]
+
+      L = {
+        'some key' => {
+          M = 2 => 3
```

Comment:
This test is getting too evolved. It represents features that I'm not keen on supporting, even if it's technically trivial at the moment.

Could we simply make a test that verifies only features that we're interested in here? That would be your initial use case:

```rb
STATUSES = [
  OPEN = 'open',
]

DISPLAY_MAPPING = {
  CANCELLED = 'cancelled' => 'Cancelled by user',
}
```